langchain/chains/query_constructor/ir.py

```diff
@@ -75,8 +75,8 @@ class Comparator(str, Enum):
     GTE = "gte"
     LT = "lt"
     LTE = "lte"
-    CONTAIN = "contain"
-    LIKE = "like"
+    CONTAIN = "list_contain"
+    LIKE = "string_pattern_like"
```

Comment:
slight concern that this will change default behavior for existing query constructor chains. wonder if we should leave enum as is but make it easier to modify the name that's actually passed into the prompt when the query constructor chain is initialized?

Reply:
Yes this is an issue we found when playing with self-query. We have opened a PR #6737 to fix this.

The major issue is that the CONTAIN comparator sometimes takes place on non-array columns and LIKE comparator sometimes appears on non-string columns. I do agree that this may change the behavior. but since these two comparators are new and added in #6143, I do think this fix may improve the prompt stability and all new features built on this may be enhanced.

Or we can leave this unchanged and open another PR for dynamic prompt for comparator?

Reply:
Reverted back and wait for future dynamic prompt PR.